.github/workflows/build.yml

```diff
@@ -97,7 +98,19 @@ jobs:
         partition: ["1/20", "2/20", "3/20", "4/20", "5/20", "6/20", "7/20", "8/20", "9/20", "10/20", "11/20", "12/20", "13/20", "14/20", "15/20", "16/20", "17/20", "18/20", "19/20", "20/20"]
 
     steps:
+
+      - name: check runner environment
+        id: check-runner
+        run: |
+          echo "name=${{ runner.name }}" >> $GITHUB_OUTPUT
+          if [[ "${{ contains( runner.name, 'arc-runners-inria' ) }}" ]]; then
+            echo "self-hosted=true" >> $GITHUB_OUTPUT
```

Comment:
"self-hosted" is kinda already available as `runner.environment == "self-hosted"`, see https://docs.github.com/en/actions/reference/workflows-and-actions/contexts#runner-context.